HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,23 @@
+import numpy as np
+from itertools import chain
+
+def P4_bfs(grph,source,sc):
+	
+	LoopBreaker = sc.accumulator(0)
+	
+	nodes = sc.parallelize([(source,0)])
+	visitedNodes = sc.parallelize([(source,0)])
+	i_dist = 0
+	while LoopBreaker.value == 0:
+		i_dist+=1
+		nodes = grph.join(nodes).values().distinct().partitionBy(8)
+		nodes = nodes.mapValues(lambda v: v+1).subtractByKey(visitedNodes)
+		if nodes.isEmpty():
+			LoopBreaker.add(1)
```

Comment:
You could use a "break" command here so your code doesn't have to run those few extra lines.